### PR TITLE
Adding CHS6540

### PR DESCRIPTION
**Pull request by Ellerbach**
## Description

Adding CHS6540

## Motivation and Context

- Adding CHS6540
- To support M5Stack Tough

## How Has This Been Tested?

Need to ne tested on a real device

## Types of changes

- [ ] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue with code or algorithm)
- [x] New feature (non-breaking change which adds functionality to code)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Unit Tests (add new Unit Test(s) or improved existing one(s), has no impact on code or features)
- [x] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist:

- [x] My code follows the code style of this project (only if there are changes in source code).
- [x] My changes require an update to the documentation (there are changes that require the docs website to be updated).
- [x] I have updated the documentation accordingly (the changes require an update on the docs in this repo).
- [x] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/main/CONTRIBUTING.md) document.
- [ ] I have tested everything locally and all new and existing tests passed (only if there are changes in source code).
- [ ] I have added new tests to cover my changes.

### CODE REVIEW

File: devices/Chs6540/Chs6540.cs [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        /// </summary>
+        public const byte DefaultI2cAddress = 0x2E;
+
+        /// <summary>
+        /// Creates a new instance of the Chs6540
+        /// </summary>
+        /// <param name="i2cDevice">The I2C device used for communication.</param>
+        public Chs6540(I2cDevice i2cDevice)
+        {
+            _i2cDevice = i2cDevice ?? throw new ArgumentException(nameof(i2cDevice));
```
Comment (josesimoes) [inline devices/Chs6540/Chs6540.cs:30]: No need to "waste" flash with the arg name. Please use the empty constructor and add a comment entry for the ArgumentException with the explanation.

File: devices/Chs6540/Chs6540.cs [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Device.I2c;
+using System.IO;
+
+namespace Iot.Device.Chs6540
+{
+    /// <summary>
+    /// Chs6540 touch screen
```
Comment (josesimoes) [inline devices/Chs6540/Chs6540.cs:11]: Missing '.' closing the sentence (and on a few others ahead).

File: devices/Chs6540/Chs6540.nfproj [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    <ProjectTypeGuids>{11A8DD76-328B-46DF-9F39-F559912D0360};{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}</ProjectTypeGuids>
+    <ProjectGuid>{F9B42E34-AD4C-4E32-89BB-FDDB9DA2AA97}</ProjectGuid>
+    <OutputType>Library</OutputType>
+    <AppDesignerFolder>Properties</AppDesignerFolder>
+    <FileAlignment>512</FileAlignment>
+    <RootNamespace>Iot.Device.Chs6540</RootNamespace>
+    <AssemblyName>Iot.Device.Chs6540</AssemblyName>
+    <TargetFrameworkVersion>v1.0</TargetFrameworkVersion>
+    <DocumentationFile>bin\$(Configuration)\Iot.Device.Chs6540.xml</DocumentationFile>
+    <LangVersion>9.0</LangVersion>
```
Comment (josesimoes) [inline devices/Chs6540/Chs6540.nfproj:19]: No need to set this here. This is already the default for our project system props.

File: devices/Chs6540/Chs6540.nuspec [hunk 4]
```diff
@@ -0,0 +1,36 @@
+<?xml version="1.0" encoding="utf-8"?>
+<package xmlns="http://schemas.microsoft.com/packaging/2012/06/nuspec.xsd">
+  <metadata>
+    <id>nanoFramework.Iot.Device.Chs6540</id>
+    <version>$version$</version>
+    <title>nanoFramework.Iot.Device.Chs6540</title>
+    <authors>nanoFramework project contributors</authors>
```
Comment (josesimoes) [inline devices/Chs6540/Chs6540.nuspec:7]:
```suggestion
    <authors>nanoframework</authors>
```

File: devices/Chs6540/Chs6540.nuspec [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    <owners>nanoFramework,dotnetfoundation</owners>
```
Comment (josesimoes) [inline devices/Chs6540/Chs6540.nuspec:8]: This property is deprecated. We're not using it anymore.
```suggestion
```

File: devices/Chs6540/README.md [hunk 6]
```diff
@@ -0,0 +1,73 @@
+# Chs6540 - Touch screen controller
+
+The Chs6540 is the touch screens controllers of the M5Stack [Tough](https://docs.m5stack.com/en/core/tough).
```
Comment (josesimoes) [inline devices/Chs6540/README.md:3]:
```suggestion
The Chs6540 is the touch screen controller of the M5Stack [Tough](https://docs.m5stack.com/en/core/tough).
```

File: devices/Chs6540/README.md [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+
+## Documentation
+
+- [More information](https://github.com/m5stack/M5Tough/blob/master/src/M5Touch.cpp) on the touch controller, unfortunately, there is datasheet available.
```
Comment (josesimoes) [inline devices/Chs6540/README.md:7]: I believe there isn't a datasheet publicly available, correct? So this should should be

```suggestion
- [More information](https://github.com/m5stack/M5Tough/blob/master/src/M5Touch.cpp) on the touch controller, unfortunately, the datasheet isn't available.
```

File: devices/Chs6540/README.md [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+//////////////////////////////////////////////////////////////////////
+// when connecting to an ESP32 device, need to configure the I2C GPIOs
+// used for the bus
+Configuration.SetPinFunction(21, DeviceFunction.I2C1_DATA);
+Configuration.SetPinFunction(22, DeviceFunction.I2C1_CLOCK);
+```
+
+For other devices like STM32, please make sure you're using the preset pins for the I2C bus you want to use.
+
+> Note: this sample requires a M5Core2.
```
Comment (josesimoes) [inline devices/Chs6540/README.md:23]:
```suggestion
> Note: this sample requires a M5Stack Tought.
```

File: devices/Chs6540/README.md [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+}
+
+void TouchInterrupCallback(object sender, PinValueChangedEventArgs pinValueChangedEventArgs)
+{
+    Debug.WriteLine("Touch interrupt");
+    var points = sensor.GetNumberPoints();
+    if (points == 1)
+    {
+        var point = sensor.GetPoint(true);
+                // Some controllers supports as well events, you can get access to them as well thru point.Event
```
Comment (josesimoes) [inline devices/Chs6540/README.md:63]:
```suggestion
        // Some controllers supports as well events, you can get access to them as well thru point.Event
```

File: devices/Chs6540/samples/Chs6540.Samples.nfproj [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+    <ProjectTypeGuids>{11A8DD76-328B-46DF-9F39-F559912D0360};{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}</ProjectTypeGuids>
+    <ProjectGuid>{070AC056-D400-4567-B04A-BA497E3C0B24}</ProjectGuid>
+    <OutputType>Exe</OutputType>
+    <AppDesignerFolder>Properties</AppDesignerFolder>
+    <FileAlignment>512</FileAlignment>
+    <RootNamespace>Iot.Device.Chs6540.Samples</RootNamespace>
+    <AssemblyName>Iot.Device.Chs6540.Samples</AssemblyName>
+    <TargetFrameworkVersion>v1.0</TargetFrameworkVersion>
+    <DocumentationFile>bin\$(Configuration)\Iot.Device.Chs6540.Samples.xml</DocumentationFile>
+    <LangVersion>9.0</LangVersion>
```
Comment (josesimoes) [inline devices/Chs6540/samples/Chs6540.Samples.nfproj:19]:
```suggestion
```

File: devices/Chs6540/version.json [hunk 11]
```diff
@@ -0,0 +1,16 @@
+{
+  "$schema": "https://raw.githubusercontent.com/dotnet/Nerdbank.GitVersioning/master/src/NerdBank.GitVersioning/version.schema.json",
+  "version": "1.0.300",
```
Comment (josesimoes) [inline devices/Chs6540/version.json:3]: OK to start at 0.

```suggestion
  "version": "1.0",
```

**Review by josesimoes — APPROVED** [review]
Great job adding this one too! Just some minor things to fix. 👍🏻 👏🏻

File: devices/Chs6540/Chs6540.cs (hunk 2 above)
Comment (Ellerbach) [reply devices/Chs6540/Chs6540.cs:11]: we'll have **a lot** of them to correct in all the repo :-) This binding is coming from FT6xxx6x which is a clone of another one (don't remember. I'll adjust but keep in mind in most of the bindings, this has no been reinforced.

File: devices/Chs6540/Chs6540.nuspec (hunk 4 above)
Comment (Ellerbach) [reply devices/Chs6540/Chs6540.nuspec:7]: shall we adjust for all the bindings? This is the standard one we are using.

File: devices/Chs6540/Chs6540.nuspec (hunk 5 above)
Comment (Ellerbach) [reply devices/Chs6540/Chs6540.nuspec:8]: shall we remove everywhere?

File: devices/Chs6540/README.md (hunk 7 above)
Comment (Ellerbach) [reply devices/Chs6540/README.md:7]: correct, there is none. So missed the not here :-D

File: devices/Chs6540/Chs6540.nuspec (hunk 4 above)
Comment (josesimoes) [reply devices/Chs6540/Chs6540.nuspec]: This has been "fixed" in all libs and several bindings. As it is not critical it's OK to make this adjustment as we touch them.

File: devices/Chs6540/Chs6540.nuspec (hunk 5 above)
Comment (josesimoes) [reply devices/Chs6540/Chs6540.nuspec]: Same as above.

**Pull request merged**